aten/src/ATen/native/Scalar.cpp

```diff
@@ -35,6 +35,9 @@ Scalar item(const Tensor& self) {
 #endif
 
 Scalar _local_scalar_dense_cpu(const Tensor& self) {
+  if (self.scalar_type() == kBool) {
+    return Scalar(static_cast<bool>(*reinterpret_cast<const uint8_t*>(self.const_data_ptr<bool>())));
```

Comment:
What's going on here?

Reply:
ASAN fails when load  a bool value out of range. Therefore, we case a bool pointer to uint8, de-reference it, and then cast to bool.

Reply:
Because of this ASAN error, many tests are marked to skip ASAN.

Reply:
A comment on this would be good